cmd/thanos/config.go

```diff
@@ -140,6 +141,9 @@ func (sc *shipperConfig) registerFlag(cmd extkingpin.FlagClause) *shipperConfig
 			"This can trigger compaction without those blocks and as a result will create an overlap situation. Set it to true if you have vertical compaction enabled and wish to upload blocks as soon as possible without caring"+
 			"about order.").
 		Default("false").Hidden().BoolVar(&sc.allowOutOfOrderUpload)
+	cmd.Flag("shipper.ignore-overlap",
+		"If true, shipper will skip checking for overlaps and allow uploading out of order blocks. Uploading out of order blocks is acceptable when vertical compaction is enabled at the compactor side.").
```

Comment:
Looks like this flag represents the same thing as the `allow-out-of-order-uploads` flag from the description. Does it make more sense to combine merge the two? Just skip the overlap checking when `allow-out-of-order-uploads` is enabled.

Reply:
Yes, I think combining them would be better.

Reply:
I think instead of skipping the check, it would be better to log the error and continue as usual if `allow-out-of-order-uploads` is enabled?

Reply:
I have used the existing flag for skipping the overlap check in a recent commit.

Reply:
Since I'm now using the existing flag, I'm working on creating a unit test for it, similar to the one for `uploadCompacted` flag.